src/modeler/AutoRest.Swagger/Validation/BodyTopLevelProperties.cs

```diff
@@ -70,11 +52,33 @@ public override IEnumerable<ValidationMessage> GetValidationMessages(Dictionary<
         /// </summary>
         public override Category Severity => Category.Error;
 
-
         /// <summary>
         /// What kind of change implementing this rule can cause.
         /// </summary>
         public override ValidationChangesImpact ValidationChangesImpact => ValidationChangesImpact.ServiceImpactingChanges;
 
+        /// <summary>
+        /// This rule passes if the model definition contains top level properties only from the allowed set: name, type,
+        /// id, location, properties, tags, plan, sku, etag, managedBy, identity
+        /// </summary>
+        /// <param name="definitions">The model definitions</param>
+        /// <param name="context">The context object</param>
+        /// <returns>validation messages</returns>
+        public override IEnumerable<ValidationMessage> GetValidationMessages(Dictionary<string, Schema> definitions, RuleContext context)
+        {
+            foreach(string resourceModel in context.ResourceModels)
+            {
+                IEnumerable<KeyValuePair<string, Schema>> topLevelProperties = ValidationUtilities.EnumerateProperties(resourceModel, definitions);
+                IEnumerable<KeyValuePair<string, Schema>> violatingProperties = topLevelProperties.Where(topLevelProperty => !AllowedTopLevelProperties.Contains(topLevelProperty.Key));
+
+                if (violatingProperties != null && violatingProperties.Any())
+                {
+                    IEnumerable<string> list = violatingProperties.Select(prop => prop.Key);
+
+                    yield return new ValidationMessage(new FileObjectPath(context.File, context.Path.AppendProperty(resourceModel).AppendProperty("properties")), this,
+                    resourceModel, string.Join(",", list));
```

Comment:
minor: you may want to add space after `","` to increase readability of the message. If it already does ad space then it's good.